EXAMPLES.md

```diff
@@ -0,0 +1,202 @@
+# Examples
+
+- [Request Options](#request-options)
+- [Pagination](#pagination)
+  - [Page based pagination](#page-based-pagination)
+  - [Checkpoint pagination](#checkpoint-pagination)
+- [Custom User Structs](#providing-a-custom-user-struct)
+
+## Request Options
+
+Fine-grained configuration can be provided on a per-request basis to enhance the request with specific query params, headers or to pass it a custom context.
+
+```go
+// Example
+userGrants, err := auth0API.Grant.List(
+	management.Context(ctx)
+	management.Header("MySpecialHeader","MySpecialHeaderValue")
+    management.Parameter("user_id", "someUserID"),
```

Comment:
When viewed as rendered Markdown (e.g. https://github.com/auth0/go-auth0/blob/24fdede2333c79dca1033c99c63b11b5fbda1972/EXAMPLES.md) the formatting seems to be off for the first two items:

<img width="584" alt="Screenshot 2023-03-14 at 14 01 07" src="https://user-images.githubusercontent.com/5055789/225081655-acad035d-ec4e-4e0b-868d-c4991eac07fd.png">